areas/storage/src/AzureMcp.Storage/Services/StorageService.cs

```diff
@@ -753,4 +753,49 @@ private static StorageSkuName ParseStorageSkuName(string sku)
             _ => throw new ArgumentException($"Invalid access tier '{accessTier}'. Valid values are: Hot, Cool")
         };
     }
+
+    public async Task<BlobUploadResult> UploadBlob(
+        string accountName,
+        string containerName,
+        string blobName,
+        string localFilePath,
+        bool overwrite,
+        string subscription,
+        string? tenant = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        ValidateRequiredParameters(accountName, containerName, blobName, localFilePath, subscription);
+
+        if (!File.Exists(localFilePath))
+        {
+            throw new FileNotFoundException($"Local file not found: {localFilePath}");
+        }
+
+        var blobServiceClient = await CreateBlobServiceClient(accountName, tenant, retryPolicy);
+        var blobContainerClient = blobServiceClient.GetBlobContainerClient(containerName);
+        var blobClient = blobContainerClient.GetBlobClient(blobName);
+
+        // Check if blob exists before upload
+        bool blobExists = await blobClient.ExistsAsync();
```

Comment:
Is this check necessary? The check and the upload are not atomic so the operation can still get errors for blob existing. Same question for the wasOverwritten parameter because it's possible the upload operation overwrites a blob without the client being aware it.